File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/BrandService.java
package com.pyg.manager.service;

import java.util.List;
import java.util.Map;

import com.pyg.pojo.PageResult;
import com.pyg.pojo.TbBrand;
import com.pyg.utils.PygResult;

public interface BrandService {

	/**
	 * 需求：查询所有品牌数据
	 */
	public List<TbBrand> findAll();

	// 返回分页列表
	public PageResult findPage(int pageNum, int pageSize);

	// 添加商品品牌
	public PygResult add(TbBrand brand);

	// 修改商品updateByprimaryKey
	public PygResult updateByprimaryKey(TbBrand brand);

	// 查询单个商品
	public TbBrand findOne(Long id);

	// 删除商品品牌
	public PygResult delete(Long[] id);

	// 品牌条件查询
	public PageResult queryBrand(TbBrand brand, int page, int rows);

	//查询,下拉框select2
	public List<Map> findBrandList();

}

File: pyg-shop-web2/src/main/webapp/js/service/brandService.js
	//绑定一个service
	app.service("brandService", function($http) {
		//查询所有商品
		this.findAll = function() {
			return $http.get("../brand/findAll");
		};

		//分页查询商品
		this.findPage = function(page, rows) {
			return $http.get("../brand/findPage?page=" + page + "&rows=" + rows);
		};

		//添加
		this.add = function(entity) {
			return $http.post("../brand/add" , entity)
		};
		//修改
		this.update = function(entity) {
			return $http.post("../brand/update" , entity)
		};

		//查询单个商品
		this.findOne = function(id) {
			return $http.get("../brand/findOne?id=" + id);
		}

		//删除商品
		this.dele = function(ids) {
			return $http.get("../brand/delete?ids=" + ids);
		}
		//select2
		this.findBrandList=function(){
			return $http.get("../brand/findBrandList");
		}

	});
	